### PR TITLE
[ADD] mail_message: added model_id field to provide search by applicatio...

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
...n.

[IMP/FIX]mail: mail.js: hide 'Mark all read' button, when thread is having all the messages makred as read.
- Task: https://www.odoo.com/web#id=6824&view_type=form&model=project.task&action=333&active_id=131
- Pad: https://pad.odoo.com/p/r.JmRhE3JrBVAPrpye

### CODE REVIEW

File: addons/mail/static/src/js/mail.js [hunk 1]
```diff
@@ -35,6 +35,20 @@ openerp.mail = function (session) {
             return [text, false];
         },
 
+        /* If there is unread message in thread, displays Mark all read button, else hide it.
+        */
+        show_mark_all_read : false,
```
Comment (tde-banana-odoo) [inline addons/mail/static/src/js/mail.js]: Ugh ? This should be a variable on the Wall or Thread manager, not on the chatter utilities ...

File: addons/mail/static/src/js/mail.js [hunk 2]
```diff
@@ -35,6 +35,20 @@ openerp.mail = function (session) {
...
+
+        check_message_read: function(){
```
Comment (tde-banana-odoo) [inline addons/mail/static/src/js/mail.js]: Same thing here. Chatter utils holds some tools methods for the chatter. This method should be a class method for the wall / thread manager.

File: addons/mail/mail_message.py [hunk 3]
```diff
@@ -188,6 +193,11 @@ def _get_default_author(self, cr, uid, context=None):
         'email_from': lambda self, cr, uid, ctx=None: self._get_default_from(cr, uid, ctx),
     }
 
+    def onchange_model(self, cr, uid, ids, model, context=None):
+        if not model:
+            return {'value': {'model_id': False}}
+        return {'value': {'model_id': self.pool['ir.model'].search(cr, uid, [('model', '=', model)], context=context)[0]}}
```
Comment (mart-e) [inline addons/mail/mail_message.py]: add `limit=1` in search

File: addons/mail/static/src/js/mail.js (hunk 1 above)
Comment (dka-odoo) [inline addons/mail/static/src/js/mail.js]: Done!

File: addons/mail/static/src/js/mail.js (hunk 2 above)
Comment (dka-odoo) [inline addons/mail/static/src/js/mail.js]: Done!

File: addons/mail/mail_message.py (hunk 3 above)
Comment (dka-odoo) [inline addons/mail/mail_message.py]: Done!

File: addons/mail/mail_message.py [hunk 4]
```diff
@@ -803,6 +813,9 @@ def create(self, cr, uid, values, context=None):
         context = dict(context or {})
         default_starred = context.pop('default_starred', False)
 
+        if 'model' in values and 'model_id' not in values:
```
Comment (vme-odoo) [inline addons/mail/mail_message.py]: optimize with values.get('model_id')

File: addons/mail/mail_message.py [hunk 5]
```diff
@@ -834,6 +847,11 @@ def read(self, cr, uid, ids, fields=None, context=None, load='_classic_read'):
         res = super(mail_message, self).read(cr, uid, ids, fields=fields, context=context, load=load)
         return res
 
+    def write(self, cr, uid, ids, values, context=None):
+        if 'model' in values and 'model_id' not in values:
```
Comment (vme-odoo) [inline addons/mail/mail_message.py]: same as above

File: addons/mail/mail_message.py [hunk 6]
```diff
@@ -188,6 +193,11 @@ def _get_default_author(self, cr, uid, context=None):
         'email_from': lambda self, cr, uid, ctx=None: self._get_default_from(cr, uid, ctx),
     }
 
+    def onchange_model(self, cr, uid, ids, model, context=None):
```
Comment (vme-odoo) [inline addons/mail/mail_message.py]: remove ids, it not used anywhere

Comment (vme-odoo) [inline addons/mail/mail_message.py]: rename function name, it hints like onchange methods

File: addons/mail/static/src/js/mail.js [hunk 7]
```diff
@@ -2025,7 +2051,9 @@ openerp.mail = function (session) {
                 'domain' : domain,
                 'context' : context,
             }));
-            return this.root.replace(this.$('.oe_mail-placeholder'));
+            var result = this.root.replace(this.$('.oe_mail-placeholder'));
+            mail.ChatterUtils.show_mark_all_read = search && search['domain'].length ? true : false;
+            return result
```
Comment (vme-odoo) [inline addons/mail/static/src/js/mail.js]: why use result, it can directly return after assign value of show_mark_all_read

Comment (dka-odoo) [inline addons/mail/static/src/js/mail.js]: Done!

File: addons/mail/mail_message.py (hunk 6 above)
Comment (dka-odoo) [inline addons/mail/mail_message.py]: removed ids :) thanks!..

**Comment by rim-odoo** [comment]
@dka-odoo Can you fix the conflicts and rebase the branch? Thanks

**Pull request closed without merging**